### ISSUE TITLE
question about multi-GPU vs single-GPU training

### ISSUE THREAD

**Issue by taoleicn** [OpenNMT/OpenNMT-tf#117]
Hi, thank you for sharing the tokenized data!

I have two questions about the training:
  1. You recommend training using 4 GPUs for [best performance](https://github.com/OpenNMT/OpenNMT-tf/tree/master/scripts/wmt#training). Does this impact the final BLEU score or just the overall training speed?

  2. Could you explain why they are [two hyper-parameters](https://github.com/OpenNMT/OpenNMT-tf/blob/master/scripts/wmt/config/wmt_ende.yml#L26-L27) `decay_steps` and `decay_step_duration` for warm-up, instead of one?

**Comment by guillaumekln** [comment]
Hello,

1. Both. The Transformer model is known to require large batch sizes to achieve the BLEU scores reported in the papers. Using more GPUs is a way to increase the effective batch size and the training speed.

2. `decay_steps_duration` is used to control the learning rate schedule depending on the batch size. For example, setting it to 2 means "increase the step used in the learning rate decay by 1 every 2 training steps". This is to compare with runs with other batch sizes whil keeping the same learning rate schedule.

Does this help you?

**Comment by taoleicn** [comment]
Hi,

Thank you for the answers.

So the 8k warmup and duration=2 is to be consistent with the paper's setup. I mean,
```  
  8 GPUs: duration=1 
  4 GPUs: duration=2
  ..
  1 GPU: duration = 8
```
The effective bath size is approximately 4096*8 in this setup.

Is the multi-GPU training synchronized ?

**Comment by taoleicn** [comment]
Also, did you notice any difference in BLEU between [4-GPU config](https://github.com/OpenNMT/OpenNMT-tf/blob/master/scripts/wmt/config/wmt_ende.yml) and [1-GPU config](https://github.com/OpenNMT/OpenNMT-tf/blob/master/scripts/wmt/config/wmt_ende_1gpu.yml)?  given the effective batch size is set similar in these two cases.

**Comment by guillaumekln** [comment]
The effective batch size is still different but the learning rate schedule is the same.

Say you have N GPUs and set the following configuration:

```yml
params:
  decay_steps_duration: 8/N
train:
  batch_size: 4096
  batch_type: tokens
```

* The parameters are updated every 4096*N tokens (this is is the effective batch size)
* The learning rate is decayed every 4096*8 tokens

Hope that makes sense.

**Comment by taoleicn** [comment]
Got it. Thank you so much!
